File: Assets/Tests/TestGetBasicValues.cs
namespace Nahoum.UnityJSInterop.Tests
{
    public class TestGetBasicValues
    {
        [ExposeWeb]
        public static string GetTestString() => "This is a test string";

        [ExposeWeb]
        public static double GetTestDouble() => 123456789.123456789;

        [ExposeWeb]
        public static int GetTestInt() => 123456789;

        [ExposeWeb]
        public static bool GetTestBoolTrue() => true;

        [ExposeWeb]
        public static bool GetTestBoolFalse() => false;

        [ExposeWeb]
        public static string GetNullString() => null;
    }
}

File: sample-unity-js-interop-ts/src/tests/test-get-basic-values.ts
import { UnityInstance } from "../UnityJSInterop";
import { SampleValues } from "../test-utilities/sample-values";
import { assertEqual } from "../test-utilities/test-asserts";

export function RunTestGetBasicValues(unityInstance: UnityInstance){
    // New tests
    assertEqual(unityInstance.Module.static.Nahoum.UnityJSInterop.Tests.TestGetBasicValues.GetTestInt().value, SampleValues.TestInt, "TestGetBasicValues.GetTestInt");
    assertEqual(unityInstance.Module.static.Nahoum.UnityJSInterop.Tests.TestGetBasicValues.GetTestDouble().value, SampleValues.TestDouble, "TestGetBasicValues.GetTestDouble");
    assertEqual(unityInstance.Module.static.Nahoum.UnityJSInterop.Tests.TestGetBasicValues.GetTestBoolFalse().value, SampleValues.TestBoolFalse, "TestGetBasicValues.GetTestBoolFalse");
    assertEqual(unityInstance.Module.static.Nahoum.UnityJSInterop.Tests.TestGetBasicValues.GetTestBoolTrue().value, SampleValues.TestBoolTrue, "TestGetBasicValues.GetTestBoolTrue");
    assertEqual(unityInstance.Module.static.Nahoum.UnityJSInterop.Tests.TestGetBasicValues.GetNullString().value, null, "TestGetBasicValues.GetNullString");
}